### DISCUSSION TITLE
How do I bind a DataTable to a DataGrid?

### DISCUSSION THREAD

**Discussion by tobeforever** [AvaloniaUI/Avalonia#13238] [Q&A] [answered]
I have a data set that needs to be displayed in the DataGrid. Because the column content is generated automatically based on the configuration, there is no way for me to construct an object in the code in advance.
My plan is to use the DataTable to place the queried data in the database, iterate through setting the caption of each column, and bind it in the DataGrid. Then I populate the DataTable dataset. But there is nothing displayed in the current DataGrid. How do I fulfill my needs?

```C#
    private DataTable _realTimeData;

    public DataTable RealTimeData
    {
        get => _realTimeData;
        set => SetProperty(ref _realTimeData, value);
    }

    private void InitDataTable()
    {
        _realTimeData = new DataTable();

        _realTimeData.Columns.Add(new DataColumn("ID", typeof(int)) { Caption = "ID" });
        _realTimeData.Columns.Add(new DataColumn("DataTime", typeof(DateTime)) { Caption = "DateTime" });

        foreach(QueryNode queryNode in CurrentQueryGroup?.queryNodes)
        {
            _realTimeData.Columns.Add(new DataColumn(queryNode.id, typeof(string)) { Caption = $"{queryNode.name}({queryNode.unitId})" });
        }
    }

```

```axaml
<DataGrid
	ItemsSource="{Binding RealTimeData}">
</DataGrid>
```

**Comment by timunie** [comment]
that's not supported by Avalonia yet. We have however `DataGridCollectionView`. Maybe also worth to check out TreeDataGrid

**Comment by Philltorian** [comment, marked as answer]
Here's a quick and nasty example of how to bind a dynamic datatable to the datagrid

Hopefully it should be enough to get you started
[AvaloniaDatagridBindingToDatatable.zip](https://github.com/AvaloniaUI/Avalonia/files/12908824/AvaloniaDatagridBindingToDatatable.zip)

**Reply by tobeforever** [reply]
Thanks. I'll try
